core/node/app_registry/cached_encrypted_message_queue.go

```diff
@@ -37,42 +35,33 @@ type SessionMessages struct {
 type CachedEncryptedMessageQueue struct {
 	store         storage.AppRegistryStore
 	appDispatcher *AppDispatcher
-	// the appIdCache stores forward settings for each app. These are used
-	// to determine whether or not to forward to each app in a channel whenever
-	// a new message is received.
-	appIdCache sync.Map
-
-	// metadataCache stores app metadata with TTL to reduce database queries.
-	metadataCache *arc.ARCCache[string, *metadataCacheEntry]
+	// appInfoCache stores complete AppInfo objects with TTL to reduce database queries.
+	// This includes settings, metadata, webhook status, and active status.
+	appInfoCache *arc.ARCCache[string, *appInfoCacheEntry]
```

Comment:
```suggestion
	appInfoCache *arc.ARCCache[common.Address, *appInfoCacheEntry]
```

`common.Address` is comparable and faster than string comparison and prevents mismatches due to case differences.